Remove vestiges of view-based `_changes` feed from codebase

eiri opened the pull request:
## Overview

This PR removes all bits and pieces of view_changes from mrview, couch_changes and fabric applications, removes `/_view_changes` end-point and removes seq, kseq and log btrees from mrview file structure. Migration code for 1.2 views was modified to support transparent migration from 2.x views.

## Testing recommendations

All the tests should pass. Sinice this is removing functionality tests for which been disabled anyway, it's hard to validate it otherwise.

Additional manual testing could be done with creation of views on CouchDB 2.x with `seq_indexed` and / or `keyseq_indexed` options set and then accessing them on this CouchDB version. Here is [a setup I've used for testing](https://gist.github.com/eiri/1e5ba550f5dbd271b0b7d03d65e66414), it could help to get up to speed with that. 

## Related Issues or Pull Requests

Closes #2167 

## Checklist

- [x] Code is written and works correctly
- [x] Changes are covered by tests
- [ ] Any new configurable parameters are documented in `rel/overlay/etc/default.ini`
- [ ] A PR for documentation changes has been made in https://github.com/apache/couchdb-documentation

kocolosk commented on src/couch_mrview/src/couch_mrview_util.erl:
```diff
@@ -1168,10 +965,10 @@ maybe_update_index_file(State) ->
     end.
 
 update_index_file(State) ->
```
Suggest upgrading the logging here to `notice` level and including more details about the index file that we're attempting to update.

eiri commented:
@kocolosk @davisp Thank you for the reviews! I've added details to upgrade logs and fixed a small upgrade bug, added a commit right after upgrade to make sure we have new sig in header on disk. Will to merge once travis allows and then add a note to upgrade docs on views 1.x incompatibility.
